video/filter/vf_vapoursynth.c

```diff
@@ -699,6 +700,8 @@ static int reinit_vs(struct priv *p, struct mp_image *input)
     p->vsapi->propSetFloat(vars, "container_fps", container_fps, 0);
     p->vsapi->propSetFloat(vars, "display_fps", display_fps, 0);
     p->vsapi->propSetIntArray(vars, "display_res", display_res, 2);
+    p->vsapi->propSetData(vars, "user_data", p->opts->user_data,
+                          strlen(p->opts->user_data), 0);
```

Comment:
How is this data accessed in VS script? Wouldn't be better to `strlen(p->opts->user_data) + 1` to copy also null terminator? Even if not used, I think it wouldn't harm.

Reply:
VapourSynth internally [converts the passed pointer and size into a C++ `std::string`](https://github.com/vapoursynth/vapoursynth/blob/dc5a5714f353649bdb32577aa35117860ded1697/src/core/vsapi.cpp#L594). Including the `\0` in size would result the string ending with an extra `\0`.

What I should do is pass a negative there so `std::string()` itself takes care of the size. Doc: http://www.vapoursynth.com/doc/api/vapoursynth4.h.html#mapsetdata

Reply:
BTW, I just noticed that mpv is still using the v3 VapourSynth API, which is deprecated. I should create another PR later to update to v4.